sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -301,12 +320,226 @@ export class IER12ProcessingService {
         totalAssessedCost: assessmentData.totalAssessedCost,
         totalAssessmentNeed: assessmentData.totalAssessmentNeed,
         disbursementSentDate: disbursement.dateSent,
+        applicationEventCode: await this.getApplicationEventCode(
+          application.applicationNumber,
+          application.applicationStatus,
+          disbursement,
+          student.studentRestrictions,
+        ),
       };
       ier12Records.push(ier12Record);
     }
     return ier12Records;
   }
 
+  /**
+   * Get application event code (i.e current state of an application)
+   * @param applicationNumber application number.
+   * @param applicationStatus application status.
+   * @param currentDisbursementSchedule current disbursement schedule.
+   * @param studentRestrictions student restrictions.
+   * @returns application event code.
+   */
+  private async getApplicationEventCode(
+    applicationNumber: string,
+    applicationStatus: ApplicationStatus,
+    currentDisbursementSchedule: DisbursementSchedule,
+    studentRestrictions?: StudentRestriction[],
+  ): Promise<ApplicationEventCode> {
+    switch (applicationStatus) {
+      case ApplicationStatus.Assessment:
+        return this.applicationEventCodeDuringAssessment(applicationNumber);
+
```

Comment:
I believe that in general, we do not leave a blank line between the `case`.
If it is for readability I would not mind having it removed also.